synapse/storage/databases/main/event_federation.py

```diff
@@ -1057,7 +1057,7 @@ def _get_connected_batch_event_backfill_results_txn(
             INNER JOIN batch_events AS c
             ON i.next_batch_id = c.batch_id
             /* Get the depth of the batch start event from the events table */
-            INNER JOIN events AS e USING (event_id)
+            INNER JOIN events AS e ON c.event_id = e.event_id
```

Comment:
Can we configure SQLite to yell about ambiguous columns as well?

Reply:
(Cross link please? 🙏 )

Reply:
It's intentional behaviour and there are test cases ensuring it picks the first one: https://www.sqlite.org/forum/forumpost/941664fc450a4f56
Mostly as a legacy compatibility thing, but RIGHT and FULL joins will cause ambiguity checking (because old SQLites didn't support them and hence no applications can be relying on this).